pkg/controller/topology_test.go

```diff
@@ -1323,6 +1323,44 @@ func TestPreferredTopologies(t *testing.T) {
 			},
 			expectError: true,
 		},
+		"allowedTopologies is subset of selected node's topology": {
```

Comment:
can you add test cases for when selectedNode is not set?

Reply:
there are already test cases for that where we will pass everything aftter flattening the list, we don;t bother about supported keys by the driver.  (for e.g. https://github.com/kubernetes-csi/external-provisioner/blob/39dcde6e61ead3a5b796cf6be6d4adb2e70c1088/pkg/controller/topology_test.go#L741).

As this PR touches the code path for waitforfirstconsumer only, so added test cases for selected node only.